### PR TITLE
Skipping slow tests on Travis

### PR DESCRIPTION
**Pull request by juliebehr**
Added ```@skipif``` to ```test_samplers()```.

Does it belong on other tests as well?

### CODE REVIEW

**Comment by juliebehr** [comment]
(I'm watching the build tests to try to tell if there are any others that are obviously holding it up)

**Comment by juliebehr** [comment]
According to the most recent completed build, this will prevent 2 of the 4 longest tests -- the other 2 are both ```test_topology_proposal``` and ```test_geometry_engine``` which take about 34 and 39 minutes in python 2.7 and 3.4 respectively.  ```test_ncmc_integrator``` and ```test_elimination``` takes roughly 13 minutes in both pythons.  All others are < 7mins.

**Comment by juliebehr** [comment]
Update: the build that is currently running is going slower; all tests besides these 6 are roughly the same:
```
2:59    - python=2.7 CONDA_PY=27 NOSETESTS="perses.tests.testsystems perses.tests.utils"
2:48    - python=2.7 CONDA_PY=27 NOSETESTS="perses.samplers"
2:51    - python=2.7 CONDA_PY=27 NOSETESTS="perses.tests.test_samplers"
2:29    - python=2.7 CONDA_PY=27 NOSETESTS="perses.tests.test_samplers" TESTSYSTEMS="ValenceSmallMoleculeLibraryTestSystem"
2:08    - python=2.7 CONDA_PY=27 NOSETESTS="perses.tests.test_samplers" TESTSYSTEMS="T4LysozymeInhibitorsTestSystem"
*1:14:00 & still running    - python=2.7 CONDA_PY=27 NOSETESTS="perses.tests.test_samplers" TESTSYSTEMS="KinaseInhibitorsTestSystem"*
4:58    - python=2.7 CONDA_PY=27 NOSETESTS="perses.tests.test_samplers" TESTSYSTEMS="AlkanesTestSystem"
7:53    - python=2.7 CONDA_PY=27 NOSETESTS="perses.tests.test_samplers" TESTSYSTEMS="AlanineDipeptideTestSystem"
*1:12:00 & still running    - python=2.7 CONDA_PY=27 NOSETESTS="perses.tests.test_samplers" TESTSYSTEMS="AblImatinibTestSystem"*
*1:09:00 & still running    - python=2.7 CONDA_PY=27 NOSETESTS="perses.rjmc perses.tests.test_topology_proposal perses.tests.test_geometry_engine"*
2:04    - python=2.7 CONDA_PY=27 NOSETESTS="perses.bias"
*14:21    - python=2.7 CONDA_PY=27 NOSETESTS="perses.annihilation perses.tests.test_ncmc_integrator perses.tests.test_elimination"*
3:16    - python=3.4 CONDA_PY=34 NOSETESTS="perses.tests.testsystems perses.tests.utils"
2:17    - python=3.4 CONDA_PY=34 NOSETESTS="perses.samplers"
*41:45    - python=3.4 CONDA_PY=34 NOSETESTS="perses.rjmc perses.tests.test_topology_proposal perses.tests.test_geometry_engine"*
2:17    - python=3.4 CONDA_PY=34 NOSETESTS="perses.bias"
*48:00 & still running    - python=3.4 CONDA_PY=34 NOSETESTS="perses.annihilation perses.tests.test_ncmc_integrator perses.tests.test_elimination"*
```

File: perses/tests/test_samplers.py [hunk 1]
```diff
@@ -68,6 +69,7 @@ def test_valence():
             f.description = "Testing MultiTargetDesign sampler with %s transfer free energy from vacuum -> %s" % (testsystem_name, environment)
             yield f
 
+@skipIf(os.environ.get("TRAVIS", None) == 'true', "Skip test samplers for now to prevent holding up Travis")
```
Comment (jchodera) [inline perses/tests/test_samplers.py]: These tests are already controllable through `.travis.yml`.  You can see on lines 81-82 that if the `TESTSYSTEMS` environment variable is set, only those tests are run.  We can just set `TESTSYSTEMS` to `ValenceSmallMoleculeLibraryTestSystem` in `.travis.yml` and omit any other tests for `perses.tests.test_samplers`.

**Comment by jchodera** [comment]
> Added @skipif to test_samplers().
> Does it belong on other tests as well?

We should be adding this to *other* tests that are slow, since `test_samplers()` already has a mechanism for restricting which tests are run via the `TESTSYSTEMS` environment variable set in the `.travis.yml` branches.

**Comment by jchodera** [comment]
This set of tests takes 40 min:
```
    - python=2.7 CONDA_PY=27 NOSETESTS="perses.rjmc perses.tests.test_topology_proposal perses.tests.test_geometry_engine"
```
Can we split [that line](https://github.com/choderalab/perses/blob/master/.travis.yml#L52) up into three separate branches?
```
    - python=2.7 CONDA_PY=27 NOSETESTS="perses.rjmc"
    - python=2.7 CONDA_PY=27 NOSETESTS="perses.tests.test_topology_proposal" 
    - python=2.7 CONDA_PY=27 NOSETESTS="perses.tests.test_geometry_engine"
```
?

**Comment by jchodera** [comment]
Same with [the py34 version of that line](https://github.com/choderalab/perses/blob/master/.travis.yml#L57), though maybe we don't even need it.

**Comment by jchodera** [comment]
[This line](https://github.com/choderalab/perses/blob/master/.travis.yml#L59) takes 14 min:
```
    - python=3.4 CONDA_PY=34 NOSETESTS="perses.annihilation perses.tests.test_ncmc_integrator perses.tests.test_elimination"
```
Maybe it can be split into individual tests as well:
```
    - python=3.4 CONDA_PY=34 NOSETESTS="perses.annihilation"
    - python=3.4 CONDA_PY=34 NOSETESTS="perses.tests.test_ncmc_integrator
    - python=3.4 CONDA_PY=34 NOSETESTS="perses.tests.test_elimination"
```

**Comment by jchodera** [comment]
So we may not even need `skipIf`...

**Comment by juliebehr** [comment]
The rate limiting step here is by far ```test_geometry_engine```, followed by ```test_elimination```.  Should any of the tests inside either of these be skipped on travis for now?

**Comment by jchodera** [comment]
Nearly all the `test_geometry_engine` time is spent in [`test_mutate_all_to_all`](https://github.com/choderalab/perses/blob/skipif/perses/tests/test_geometry_engine.py#L137-L142): 1798 seconds!  That's 400 transformations.  Not sure if there's a way to speed that up...

**Comment by jchodera** [comment]
We should probably mark `test_mutate_all_to_all` in `test_geometry_engine` to be skipped on travis...

**Comment by juliebehr** [comment]
Anything to leave out of ```test_elimination```, or is this acceptable?

**Comment by juliebehr** [comment]
Build completed in 37 minutes.

**Comment by jchodera** [comment]
I seriously newd to turn off a lot of that debug printing in test_elimination. Any idea what the slow parts are? 

I think this is a huge improvement already. Can we commit this and then I can fix up test_elimination?

**Comment by juliebehr** [comment]
I can't really tell from going through the tests, because it doesn't seem to give me # of seconds except for when taking 0 steps and 1 step, but I think all of the 50 steps are failing.  And yes!

**Pull request merged**